### PR TITLE
Minor: Improve doc comments in FileStream

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/5788.
Closes https://github.com/apache/arrow-datafusion/pull/5790

# Rationale for this change

It is clear that the metrics are confusing but they are still useful as described by @thinkharderdev  in https://github.com/apache/arrow-datafusion/pull/5790#pullrequestreview-1368680282

# What changes are included in this PR?
1. Add doc comments to try and explain in more detail what the metrics mean. 
 
# Are these changes tested?
N/A
# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/src/physical_plan/file_format/file_stream.rs [hunk 1]
```diff
@@ -140,14 +141,31 @@ impl StartableTime {
...
+    /// Time between when the [`FileStream`] requests data from the
+    /// stream and when the first [`RecordBatch`] is produced.
     pub time_scanning_until_data: StartableTime,
-    /// Total elapsed time for for scanning + record batch decompression / decoding
+    /// total elapsed wall clock time for for scanning + record batch decompression / decoding
+    ///
+    /// Sum of time between when the [`FileStream`] requests data from
+    /// the stream and when a [`RecordBatch`] is produced for all
+    /// record batches in the stream.
     pub time_scanning_total: StartableTime,
```
Comment (yahoNanJing) [inline datafusion/core/src/physical_plan/file_format/file_stream.rs:165]: This time may include the time of the parent operator's execution. It's better to indicate this info. And maybe it's better to change `time_scanning_total` to other names.

Comment (alamb) [reply datafusion/core/src/physical_plan/file_format/file_stream.rs:166]: Updated in 94ecfc49fa

**Comment by alamb** [comment]
@thinkharderdev  and @nenorbot since you recently looked at this code I wonder if you have some time to review this PR (it adds comments, no code changes)

File: datafusion/core/src/physical_plan/file_format/file_stream.rs [hunk 2]
```diff
@@ -140,14 +141,32 @@ impl StartableTime {
...
+    /// [`FileStream`] receives a stream for reading.
     pub time_opening: StartableTime,
-    /// Time elapsed for file scanning + first record batch of decompression + decoding
+    /// Wall clock time elapsed for file scanning + first record batch of decompression + decoding
+    ///
+    /// Time between when the [`FileStream`] requests data from the
+    /// stream and when the first [`RecordBatch`] is produced.
     pub time_scanning_until_data: StartableTime,
-    /// Total elapsed time for for scanning + record batch decompression / decoding
+    /// total elapsed wall clock time for for scanning + record batch decompression / decoding
```
Comment (waynexia) [inline datafusion/core/src/physical_plan/file_format/file_stream.rs:160]: nit:
```suggestion
    /// Total elapsed wall clock time for for scanning + record batch decompression / decoding
```

**Review by waynexia — APPROVED** [review]
Thank you for providing such clear documents 👍

File: datafusion/core/src/physical_plan/file_format/file_stream.rs [hunk 3]
```diff
@@ -140,14 +141,32 @@ impl StartableTime {
...
+///
+/// Note that all of these metrics are in terms of wall clock time
+/// (not cpu time) so they include time spent waiting on I/O as well
+/// as other operators.
 struct FileStreamMetrics {
-    /// Time elapsed for file opening
+    /// Wall clock time elapsed for file opening.
+    ///
+    /// Time between when [`FileReader::open`] is called and when the
+    /// [`FileStream`] receives a stream for reading.
```
Comment (thinkharderdev) [inline datafusion/core/src/physical_plan/file_format/file_stream.rs:153]:
```suggestion
    /// Wall clock time elapsed for file opening.
    ///
    /// Time between when [`FileReader::open`] is called and when the
    /// [`FileStream`] receives a stream for reading.
    ///
    /// If there are multiple files being scanned, the stream
    /// will open the next file in the background while scanning the 
    /// current file. This metric will only capture time spent opening
    /// while not also scanning. 
```

**Review by thinkharderdev — APPROVED** [review]
One small suggestion to clarify `time_opening` but this looks great!

**Pull request merged**